docs/DeveloperGuide.md

```diff
@@ -1,38 +1,253 @@
+3. [Implementation](#implementation)
+4. [Appendix: Requirements](#appendix--requirements)
+5. [Appendix: Instructions for manual testing](#appendix--instructions-for-manual-testing)
+
+---
 
+## Acknowledgements
+x
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
-## Design & implementation
+{list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the
+original source as well}
+
+- [addressbook-level2](https://github.com/se-edu/addressbook-level2)
+- [addressbook-level3](https://github.com/se-edu/addressbook-level3)
+
+---
+
+## Design
+
+### Architecture
+
+![Architecture Diagram](images/ArchitectureDiagram.png)
+
+The ***architecture diagram*** given above is explains the high level design of the program.
+
+Given below is a quick overview of the main components and how they interact with each other.
+
+**Main components of architecture**
+
+`BrokeMan` has one class [`Main`](https://github.com/AY2223S2-CS2113-F13-2/tp/blob/master/src/main/java/seedu/brokeMan/BrokeMan.java), which is responsible for:
+- At program launch: Initialises the components in the correct sequence, and connect them up with each other
+- At program termination: Shuts down the components and invokes cleanup methods where necessary.
+
+[`Common`](#common-class) represents a collection of messages used by multiple other components.
+
+The rest of the program consists of mainly 5 main components.
+- [`Ui`](#ui-component): The Ui of the program.
+- [`Parser`](#parser-component): The user input parser.
+- [`Command`](#command-component): The command executor.
+- [`Storage`](#storage-component): Reads data from, and writes data to hard disk.
+- [`EntryList`](#entrylist-component): Stores the list of entries when program is running.
+
+### Ui component
+
+### Parser component
+
+### Command component
+
+### Storage component
+
+### EntryList component
+
+### Common class
+
+Messages used by multiple components are in the `seedu.brokeMan.commmon` package.
+
+---
+
+## implementation
+
+### Entry
+
+Entry class is the underlying superclass for Expense and Income classes. It establishes the common attributes and
+methods that are necessary to represent Expenses and Incomes. Abstract class is used to represent their common features
+to maximize code reusability and increase maintainability.
+
+Private attributes
+
+Info: String that stores the description of the entry
+
+Amount: Double that stores the monetary value of entry
+
+Time: LocalDateTime that stores the date and time of entry
+
+Category: Category that stores the type tag of entry
+
+**Methods**
+
+Getters can be used to provide the private attributes to other classes
+
+editDescription(), editAmount(), editTime(), editCategory()
+
+* Takes in corresponding parameters to edit the private attributes.
+* Used by EntryList to make edits
+
+isSameMonth()
```

Comment:
Nice work having a short summary as a description, can consider bolding "Private attributes" to make it aligned in formatting with the "Methods" that you already bolded below. 
Can also consider adding bullets to indent the subpoints "Info, Amount, Time, Category" to increase readability!
Alternatively, maybe you could use a table as another way to display these details :)